1-js/01-getting-started/3-code-editors/article.md

```diff
@@ -1,46 +1,46 @@
 
-An IDE loads the project (which can be many files), allows navigation between files, provides autocompletion based on the whole project (not just the open file), and integrates with a version management system (like [git](https://git-scm.com/)), a testing environment, and other "project-level" stuff.
+Eine IDE lädt das Projekt (das aus vielen Dateien bestehen kann), erlaubt die Navigation zwischen Dateien, stellt eine Autovervollständigung für das gesamte Projekt (und nicht nur für die geöffnete Datei) bereit und ist mit einem Versionsverwaltungssystem (wie z.B.: [git](https://git-scm.com/)), einer Test-Umgebung und anderen Projektrelevanten Bereichen verknüpft.
 
-If you haven't selected an IDE yet, consider the following options:
+Falls du für dich noch keine IDE ausgesucht hast, sind die nachfolgenden Optionen empfehlenswert:
 
-- [Visual Studio Code](https://code.visualstudio.com/) (cross-platform, free).
-- [WebStorm](http://www.jetbrains.com/webstorm/) (cross-platform, paid).
+- [Visual Studio Code](https://code.visualstudio.com/) (Cross-Plattform, kostenlos).
+- [WebStorm](http://www.jetbrains.com/webstorm/) (Cross-Plattform, ksotenpflichtig).
 
-For Windows, there's also "Visual Studio", not to be confused with "Visual Studio Code". "Visual Studio" is a paid and mighty Windows-only editor, well-suited for the .NET platform. It's also good at JavaScript. There's also a free version [Visual Studio Community](https://www.visualstudio.com/vs/community/).
+Für Windows steht außerdem "Visual Studio" zur Verfügung, diese IDE sollte nicht mit "Visual Studio Code" verwechselt werden. "Visual Studio" ist ein kostenpflichtiger und umfangreicher Editor, der insbesondere für die Arbeit mit der .Net-Plattform geeignet ist. "Visual Studio" ist außerdem gut für Javascript geeignet. Eine kostenlose Version ist unter folgender Adresse verfügbar: [Visual Studio Community](https://www.visualstudio.com/vs/community/).
 
-Many IDEs are paid, but have a trial period. Their cost is usually negligible compared to a qualified developer's salary, so just choose the best one for you.
+Viele IDEs sind kostenpflichtig, stellen aber kostenlose Test-Versionen zur Verfügung. Die Kosten sind im Verhältnis zum Gehalt eines qualifizierten Programmierers vernachlässigbar, also such einfach die Plattform aus, die dir am besten passt.
 
-## Lightweight editors
+## Quelltext-Editoren
 
-"Lightweight editors" are not as powerful as IDEs, but they're fast, elegant and simple.
+Quelltext-Editoren, im englischen auch "lightweight editors", also leichtgewichtige Editoren genannt, sind nicht so mächtig wie IDEs, aber dafür sind sie schnell, eleganz und einfach in der Nutzung.
```

Comment:
```suggestion
Quelltext-Editoren, im englischen auch "lightweight editors", also leichtgewichtige Editoren genannt, sind nicht so mächtig wie IDEs, aber dafür sind sie schnell, elegant und einfach in der Nutzung.
```

Reply:
It is a translation and not a dictionary. We should stick to the source as much as possible and stop inventing sentences. I'm not so happy with "Quelltext-Editor" too. Because the author didn't speak about source code editors, he speaks about editors with less functionality. Maybe we can find something better together.

```suggestion
"Quelltext-Editoren" sind nicht so leistungsstark wie IDEs, aber sie sind schnell, elegant und einfach.
```

Reply:
> I'm not so happy with "Quelltext-Editor" too. Because the author didn't speak about source code editors, he speaks about editors with less functionality.

Imho, given the examples at the end of the section, 'source code editors' actually captures the idea fairly well: These are bare-metal text editing tools that come with syntax highlighting and elementary (block) structure checks, clearly tailored to edit source code. In German, 'Code-Editoren' would be acceptable but that conflicts with the title of the article. An alternative might be 'Entwickler-Editoren' or simply 'Editoren'.

To complicate things further, I seems uncommon in German to denote an IDE as an 'Editor'. Thus the contrast between 'IDE' and 'lightweight editor', the latter meaning 'editor with less functionality', would not carry over to German.

Please give a note if you think that discussing translation options at this level of detail actually hampers the progress of the translation effort.